File: DONGUK/프로그래머스_WEEK 9/점프와 순간이동.py
def solution(n):
    ans = 0

    while True:
        if n == 1:
            ans += 1
            break
            
        else:
            if n%2 == 0:
                n = n / 2
            
            else:
                ans += 1
                n = (n-1) / 2

    return ans

File: JINSEOP/프로그래머스_WEEK9/점프와 순간 이동.cpp
#include <iostream>
using namespace std;

int solution(int n)
{
    int answer = 0;
    while (n != 0) {
        if ((n - 1) % 2 == 0) {
            n -= 1;
            answer++;
        }
        n /= 2;
    }
    return answer;
}
